src/phes_base.cpp

```diff
@@ -280,93 +298,148 @@ ExistingReservoir get_existing_reservoir(string name) {
-    SHPGetInfo(SHP, &nEntities, NULL, NULL, NULL);
 
-    SHPObject *shape;
-    for(int i = 0; i<nEntities; i++){
-      shape = SHPReadObject(SHP, i);
-      if (shape == NULL) {
-        fprintf(stderr, "Unable to read shape %d, terminating object reading.\n", i);
-      }
-      int idx = -1;
-      for (uint r = 0; r < reservoirs.size(); r++) {
-        if (reservoirs[r].identifier == names[i]) {
-          idx = r;
+  for (string filename : filenames) {
+    if (!file_exists(filename)) {
+      search_config.logger.error("No file: " + filename);
+      throw(1);
+    }
+    SHPHandle SHP = SHPOpen(convert_string(filename), "rb");
+    DBFHandle DBF = DBFOpen(convert_string(filename), "rb");
+    bool tiled_bluefield = use_tiled_bluefield && SHP->nShapeType == SHPT_POLYGON;
+    bool tiled_river = use_tiled_rivers && SHP->nShapeType == SHPT_ARC;
+    bool csv_names = !tiled_bluefield && !tiled_river;
+    int dbf_field = 0, dbf_name_field = 0, dbf_elevation_field = 0;
+    if (tiled_river) {
+      dbf_field = DBFGetFieldIndex(DBF, string("DIS_AV_CMS").c_str());
+      dbf_name_field = DBFGetFieldIndex(DBF, string("River_name").c_str());
+    }
+    if (tiled_bluefield) {
+      dbf_field = DBFGetFieldIndex(DBF, string("Vol_total").c_str());
+      dbf_elevation_field = DBFGetFieldIndex(DBF, string("Elevation").c_str());
+      dbf_name_field = DBFGetFieldIndex(DBF, string("Lake_name").c_str());
+    }
+    if (SHP != NULL) {
+      int nEntities;
+      SHPGetInfo(SHP, &nEntities, NULL, NULL, NULL);
+
+      for (int i = 0; i < nEntities; i++) {
+        SHPObject *shape;
+        shape = SHPReadObject(SHP, i);
+        if (shape == NULL) {
+          fprintf(stderr, "Unable to read shape %d, terminating object reading.\n", i);
+        }
+        ExistingReservoir reservoir;
+        if (csv_names) {
+          int idx = -1;
+          for (uint r = 0; r < reservoirs.size(); r++) {
+            if (reservoirs[r].identifier == names[i]) {
+              idx = r;
+            }
+          }
+          if (idx < 0) {
+            search_config.logger.debug("Could not find reservoir with id " + names[i]);
+            throw 1;
+          }
+          reservoir = reservoirs[idx];
+        } else if (tiled_bluefield) {
+          double volume = DBFReadDoubleAttribute(DBF, i, dbf_field);
+          int elevation = DBFReadIntegerAttribute(DBF, i, dbf_elevation_field);
+          string name = string(DBFReadStringAttribute(DBF, i, dbf_name_field));
+          reservoir = ExistingReservoir_init(name, 0, 0, elevation, volume);
+          reservoirs.push_back(reservoir);
+        } else if (tiled_river) {
+          double volume = DBFReadDoubleAttribute(DBF, i, dbf_field);
+          string name = string(DBFReadStringAttribute(DBF, i, dbf_name_field));
+          reservoir = ExistingReservoir_init(name, 0, 0, 0, volume);
+          reservoir.river = true;
+          reservoirs.push_back(reservoir);
```

Comment:
Why do we create the reservoirs vector and push back the reservoir objects into it for tiled_bluefield and tiled_river? The reservoirs vector doesn't seem to be used at all in these two cases

Reply:
Fixed